mantidproject/mantid
# Added tests for elemental analysis data file

Added tests to check that data file do not contain obviously wrong data (eg. negative mass) and are formatted correctly.
This includes:
 - having all peak types assigned to an element (see  #26430)
 - ensuring each element has an entry for `A`, `Z`, `Primary`, `Secondary`
 - ensuring at least one element has an entry for `Gammas`, `Electrons`

**To test:**

- Code review
- Check that the test pass with the default data file
- Modify the values of some peaks to negative ones, it should fail
- Add an entry for a non existing element, it should fail
- Remove `Pb->Electrons`, it should fail
- Remove one of `A`, `Z`, `Primary`, `Secondary` from any element, it should fail

Fixes #26593.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
